modules/local/gatk4/denoisereadcounts/main.nf

```diff
@@ -0,0 +1,54 @@
+process GATK4_DENOISEREADCOUNTS {
+    tag "$meta.id"
+    label 'process_high'
+
+    conda "bioconda::gatk4=4.2.4.1"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/gatk4:4.2.4.1--hdfd78af_0' :
+        'quay.io/biocontainers/gatk4:4.2.4.1--hdfd78af_0' }"
+
+    input:
+    tuple val(meta), path(read_counts)
+    path panel_of_normals
+
+    output:
+    tuple val(meta), path('*.standardizedCR.tsv'), emit: standardized_read_counts
+    tuple val(meta), path('*.denoisedCR.tsv')    , emit: denoised_read_counts
+    path  "versions.yml"                         , emit: versions
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def avail_mem = 12
+    if (!task.memory) {
+        log.info '[GATK DenoiseReadCounts] Available memory not known - defaulting to 12GB. Specify process memory requirements to change this.'
+    } else {
+        avail_mem = task.memory.giga
```

Comment:
Recently Matthias put in quite some effort to work on this avail mem thing for GATK to avoid a bunch of issues with docker and garbage collection and what not. Maybe take a look at any of the gatk no-core modules about the latest state there